src/main.rs

```diff
@@ -0,0 +1,83 @@
+use bytes::{ Bytes, BytesMut};
+use std::net::SocketAddr;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_stream::StreamExt;
+use tokio_util::codec::Framed;
+use futures::SinkExt;
+use anyhow::Result;
+use clap::App;
+
+pub mod tokio_netstring;
+use tokio_netstring::NetstringCodec;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let matches = App::new("simple").args_from_usage("-s --server 'Server mode'").get_matches();
+    let addr = "127.0.0.1:20000".to_string();
+
+    if matches.is_present("server") {
+        let socket = TcpListener::bind(&addr).await?;
+        loop {
+            let (connection, clientaddr) = socket.accept().await?;
+            // let mut transport = tokio_util::codec::Framed::new(connection, tokio_netstring::NetstringCodec);    
+            tokio::spawn(async move {
+                handle_connection(connection, clientaddr).await;
+            });
+        }
+    } else {
+        let connection = TcpStream::connect(&addr).await?;
+        let mut transport = tokio_util::codec::Framed::new(connection, tokio_netstring::NetstringCodec);
+        transport.send(bytes::Bytes::from("Hello World")).await?;
+    }
+
+    Ok(())
+}
+
+async fn server_process(transport: &mut Framed<TcpStream, NetstringCodec>, m: &Bytes) -> Result<()> {
+    let mut buf = BytesMut::with_capacity(m.len() + 20);
+    buf.extend_from_slice(&m);
+    buf.extend_from_slice(b" <- got this, i am echoing!");
+    transport.send(buf.freeze()).await?;
+    Ok(())
+}
+
+async fn handle_connection(connection: TcpStream, connection_address: SocketAddr) {
+    println!("conenction from {}, handling", connection_address);
```

Comment:
I'd recommend `eprintln!` for logging, as `STDOUT` should be reserved for user-consumable output.

Reply:
Also, connection is misspelled.

Reply:
all println!'s and such are to be replaced later with proper logging, like either fern module or tokio::tracing, first issue is about the busy wait on connection i need to resolve